src/components/TVChartContainer/TVChartContainer.tsx

```diff
@@ -0,0 +1,142 @@
+        const { open, liquidation } = position;
+        lines.push(drawLineOnChart(open.title, open.price));
+        lines.push(drawLineOnChart(liquidation.title, liquidation.price));
+      });
+      currentOrders.forEach((order) => {
+        lines.push(drawLineOnChart(order.title, order.price));
+      });
+    }
+    return () => {
+      lines.forEach((line) => line?.remove());
+    };
+  }, [currentPositions, savedShouldShowPositionLines, currentOrders, drawLineOnChart]);
+
+  useEffect(() => {
+    if (chartReady && tvWidgetRef.current && symbol !== tvWidgetRef.current?.activeChart?.().symbol()) {
+      tvWidgetRef.current.setSymbol(symbol, tvWidgetRef.current.activeChart().resolution(), () => {});
+    }
+  }, [symbol, chartReady, period]);
+
+  useEffect(() => {
+    if (!chartReady || !tvWidgetRef.current || !isMobile) return;
+
+    if (tvWidgetRef.current.activeChart().getCheckableActionState("drawingToolbarAction")) {
+      tvWidgetRef.current?.activeChart().executeActionById("drawingToolbarAction");
+    }
+  }, [isMobile, chartReady]);
+
+  useEffect(() => {
+    const widgetOptions = {
+      debug: false,
+      symbol: symbol,
+      datafeed: datafeed,
+      theme: defaultChartProps.theme,
+      container: chartContainerRef.current,
+      library_path: defaultChartProps.library_path,
+      locale: defaultChartProps.locale,
+      loading_screen: defaultChartProps.loading_screen,
+      enabled_features: defaultChartProps.enabled_features,
+      disabled_features: isMobile
+        ? defaultChartProps.disabled_features.concat(disabledFeaturesOnMonile)
+        : defaultChartProps.disabled_features,
+      client_id: defaultChartProps.clientId,
+      user_id: defaultChartProps.userId,
+      fullscreen: defaultChartProps.fullscreen,
+      autosize: defaultChartProps.autosize,
+      custom_css_url: defaultChartProps.custom_css_url,
+      overrides: defaultChartProps.overrides,
+      interval: getPeriodFromResolutions(period),
+      save_load_adapter: new SaveLoadAdapter(chainId, tvCharts, setTvCharts, onSelectToken),
+    };
+    tvWidgetRef.current = new window.TradingView.widget(widgetOptions);
+
+    tvWidgetRef.current?.onChartReady(function () {
+      setChartReady(true);
+      tvWidgetRef.current?.applyOverrides({
+        "paneProperties.background": "#16182e",
+        "paneProperties.backgroundType": "solid",
+      });
+      tvWidgetRef.current
+        ?.activeChart()
+        .onIntervalChanged()
+        .subscribe(null, (interval) => {
+          if (supportedResolutions[interval]) {
+            const period = supportedResolutions[interval];
+            setPeriod(period);
+          }
+        });
+    });
+
+    return () => {
+      if (tvWidgetRef.current) {
+        tvWidgetRef.current.remove();
+        tvWidgetRef.current = null;
+        setChartReady(false);
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
this hook runs only a the first render, so, datafeed will be set only once, Is that okay?

Reply:
Yes! Otherwise chart will re-load again and again.